### DISCUSSION TITLE
https://klp1.com/issues/fan-speed/

### DISCUSSION THREAD

**Discussion by giscus** [blakadder/klp1#6] [Announcements]
# https://klp1.com/issues/fan-speed/

Enable variable speed on stock part cooling fan

https://klp1.com/issues/fan-speed/

**Comment by odsilo** [comment]
Well, at first is looks ok!, but no.. speed is very jerky .. I tried lowering cycle_time to 0.03333 and that helped a little but its not good!

**Reply by waintdeir** [reply]
Hi, maybe i have a newer version but on my printer the fan works fine and the stock values are only:
>[fan]
>pin:MKS_THR:gpio1
>cycle_time: 0.0010

Maybe this helps someone.

**Reply by odsilo** [reply]
It is the fan, that's the problem. The fan that isn't working is 24volt 0.25A and from a company called
 " xuhongxiang ". I got a new one from Kingroon, It just says 24volt... no amp. and that works flawless 👍🏻

**Reply by yarabanov** [reply]
Hello, I have problems after updating this with starting my printer. Can someone share the default configuration ot the printer.cfg and MKS_THR.cfg as no matter what I do Klipper cannot start. I get different errors like:
mcu 'MKS_THR': Unable to connect
Once the underlying issue is corrected, use the
"FIRMWARE_RESTART" command to reset the firmware, reload the
config, and restart the host software.
Error configuring printer
Wrote to Kingroon, but nobody responds there. Thanks if someone helps.
Cheers,
Yordan

**Comment by yarabanov** [comment]
Hello, I have problems after updating this with starting my printer. Can someone share the default configuration ot the printer.cfg and MKS_THR.cfg as no matter what I do Klipper cannot start. I get different errors like:
mcu 'MKS_THR': Unable to connect
Once the underlying issue is corrected, use the
"FIRMWARE_RESTART" command to reset the firmware, reload the
config, and restart the host software.
Error configuring printer
Wrote to Kingroon, but nobody responds there. Thanks if someone helps.
Cheers,
Yordan

**Reply by DPG74** [reply]
#####################################################################
# 	KINGROON KLP1 V2.2
####################################################################
[mcu MKS_THR]
serial:/dev/ttyS0
restart_method: command
#####################################################################
#              风扇
#####################################################################


 [fan]
pin:MKS_THR:gpio3
cycle_time: 0.0010



[heater_fan Fan1]
pin: MKS_THR:gpio11
shutdown_speed: 1


#####################################################################
#   自动调平和偏移
#####################################################################
[bed_mesh]
speed:150                #调平速度
horizontal_move_z:8   #调平中Z抬高高度
mesh_min:20,20        #探测点最小位置
mesh_max:202,190      #探测点最大位置
probe_count:6,6       #调平点数

**Reply by DPG74** [reply]
This is the original MKS_THR.cfg file

**Comment by longpole001** [comment]
i am also having this problem , where when the MKS_THR.cfg , is edited to change to the first post , for the fan , gives same error "MKS_THR" unable to connect " , changing the fan section back to the default , gives the same error , wonder if the file updated writes it badly , or printer not like any update to this file and wont allow it to beset , 

can the orginal,file be copied back to the printer directly , and can the orginal file be uploaded here so it can be uploaded directly to the printer , 


cant see why the printer is not happy with the orginal edited file , even when the command are placed back from the orginal txt
